api/server/server_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package server
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/snow"
+)
+
+func TestRejectMiddleware(t *testing.T) {
+	type test struct {
+		name            string
+		handlerFunc     func(*require.Assertions) http.Handler
+		contextFunc     func() *snow.ConsensusContext
+		expectedErrCode int
+	}
+
+	tests := []test{
+		{
+			name: "chain is bootstrapping",
+			handlerFunc: func(require *require.Assertions) http.Handler {
+				return http.HandlerFunc(func(_ http.ResponseWriter, _ *http.Request) {
+					require.Fail("shouldn't have called handler")
+				})
+			},
+			contextFunc: func() *snow.ConsensusContext {
+				ctx := &snow.ConsensusContext{}
+				ctx.State.Set(snow.EngineState{
+					State: snow.Bootstrapping,
+				})
+				return ctx
+			},
+			expectedErrCode: http.StatusServiceUnavailable,
+		},
+		{
+			name: "chain is done bootstrapping",
+			handlerFunc: func(*require.Assertions) http.Handler {
+				return http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+					w.WriteHeader(http.StatusTeapot)
+				})
+			},
+			contextFunc: func() *snow.ConsensusContext {
```

Comment:
(No action required) Maybe just define a state value per test instead of a context function, since that's the only thing that's changing?